src/colorizer/CanvasWithOverlay.ts

```diff
@@ -1,38 +1,64 @@
-import { Vector2 } from "three";
+import { Color, Vector2 } from "three";
```

Comment:
Okay, including a note here about the rendering order.

Down at the bottom there's a `render()` method that renders the header, footer, and main viewport.
1. The header is relatively simple, just the text name of the dataset.
2. The viewport is drawn directly as an image in the canvas.
3. The footer is the most complicated.
  1. If it's enabled and we're in Export mode, it gets the dimensions of the feature legend (either numeric or categorical) and the dimensions of the overlay box (which contains the scale bar and timestamp). It then draws a white area of the screen large enough to contain both, and then draws the legend + overlay on top of the area.
  2. If it's disable or we're not in Export mode, it just renders the overlay box a and the scale bar/timestamp.

I would recommend you look at `canvas_utils` before reviewing this file to understand the helper methods I'm using--  there's only two and they help set up rendering for text.